pkg/controller/eventing/inmemory/clusterchannelprovisioner/reconcile.go

```diff
@@ -61,6 +61,11 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	ctx := context.TODO()
 	logger := r.logger.With(zap.Any("request", request))
 
+	// The reconcile requests will include a namespace if they are triggered because of changes to the
+	// objects owned by this ClusterChannelProvisioner (e.g k8s service). Since ClusterChannelProvisioner is
+	// cluster-scoped we need to unset the namespace or otherwise the provisioner object cannot be looked up.
```

Comment:
Can you add a reference to https://github.com/kubernetes-sigs/controller-runtime/issues/214 in this comment?

Reply:
Done